src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -858,37 +867,32 @@ private function verifyArrayType($type, array $value, array &$invalidTypes, $lev
     {
         $type = substr($type, 0, -2);
 
-        $suffix = '[]';
-        while (\strlen($suffix) <= $level * 2) {
-            $suffix .= '[]';
-        }
-
         if ('[]' === substr($type, -2)) {
             $success = true;
             foreach ($value as $item) {
                 if (!\is_array($item)) {
-                    $invalidTypes[$this->formatTypeOf($item, null).$suffix] = true;
+                    $invalidTypes[$this->formatTypeOf($item, null)] = true;
 
-                    return false;
-                }
-
-                if (!$this->verifyArrayType($type, $item, $invalidTypes, $level + 1)) {
+                    $success = false;
+                } elseif (!$this->verifyArrayType($type, $item, $invalidTypes, $level + 1)) {
                     $success = false;
                 }
             }
 
             return $success;
         }
 
+        $valid = true;
+
         foreach ($value as $item) {
             if (!self::isValueValidType($type, $item)) {
-                $invalidTypes[$this->formatTypeOf($item, $type).$suffix] = $value;
+                $invalidTypes[$this->formatTypeOf($item, $type)] = $value;
 
-                return false;
+                $valid = false;
```

Comment:
same here

Reply:
the same comment https://github.com/symfony/symfony/pull/30442#discussion_r265045300